helix-term/src/main.rs

```diff
@@ -117,9 +117,9 @@ FLAGS:
     setup_logging(args.verbosity).context("failed to initialize logging")?;
 
     // Before setting the working directory, resolve all the paths in args.files
-    for (path, _) in args.files.iter_mut() {
-        *path = helix_stdx::path::canonicalize(&path);
-    }
+    args.files.iter_mut().for_each(|(path, _)| {
+        *path = helix_stdx::path::canonicalize(&*path);
+    });
```

Comment:
What's the lint that fails here? I'd be surprised to see clippy prefer `for_each` to `for`. Alternatively we should be able to write this as

```rust
for (path, _) in &mut args.files {
    *path = helix_stdx::path::canonicalize(&path);
}
```